src/ConfigSetup.cpp

```diff
@@ -171,6 +176,22 @@ bool ConfigSetup::checkBool(std::string str)
   exit(EXIT_FAILURE);
 }
 
+//Same as checkBool but doesn't exit with an error if the string doesn't
+//represent a boolean value. Use for cases where we aren't sure the argument
+//string should be a boolean.
+bool ConfigSetup::isBool(std::string str)
+{
+  //short circuit for long strings
+  if (str.length() > 5) return false;
+
+  // capitalize string
+  for(uint k = 0; k < str.length(); k++) {
+    str[k] = toupper(str[k]);
+  }
+
+  return (str == "ON" || str == "TRUE" || str == "YES" || str == "OFF" || str == "FALSE" || str == "NO");
+}
+
```

Comment:
I dont think we need a separate function. If we expect to read bool but user specified a keyword other that we supported (yes, no, true, false, on, off), we should exit and with warning, as we do in `CheckBool(string)`

Reply:
If we have the following line of input

Checkpoint _argument_

we don't know until we check that single argument if it is a boolean or a filename. If it is a filename, we assume true and proceed. So, there isn't a clear expectation of a boolean value in this case.

Reply:
This would make GOMC act differently from what we have. For example, in all other place, we read a boolean as second term, if it was true, then we will expect a third argument (file name). This is how the logic of GOMC config parser is written. If we want to change that, then that's another story.

Reply:
I'm talking about how GOMC works without the patch. If you enter

Checkpoint filename

it currently assumes true and works. If you don't want that behavior, of course your change is preferred.

Reply:
Basically the difference is
1.
What I have in mine is:
`Checkpoint false`
`Checkpoint true  filename`
`Checkpooint true` => Error: file name is not specified
`Checkpooint filename` => Error: filename is not acceptable as boolean

2.
What I have in mine is:
`Checkpoint false`
`Checkpoint true  filename`
`Checkpooint true` => Error: file name is not specified
`Checkpooint filename`

Reply:
I agree. Let's get some input on which option to go with. It's easy to support either option once we know.